types/index.d.ts

```diff
@@ -14,10 +14,13 @@ export class Kafka {
   logger(): Logger
 }
 
+export type BrokersFunction = () => Promise<string[]>;
```

Comment:
The tests seem to indicate that this could also be a non-async function, so a more correct type would be 

```ts
export type BrokersFunction = () => string[] | Promise<string[]>
```

Reply:
Indeed, fixed